src/pydargs/__init__.py

```diff
@@ -89,8 +157,16 @@ def _create_object(tp: Type[Dataclass], namespace: Namespace, prefix: str = "")
     return tp(**args)
 
 
-def _create_parser(tp: Type[Dataclass], **kwargs: Any) -> ArgumentParser:
+def _create_parser(tp: Type[Dataclass], load_from_config: bool, **kwargs: Any) -> ArgumentParser:
     parser = ArgumentParser(**kwargs, argument_default=SUPPRESS)
+    if load_from_config:
+        supported_types = "JSON- or YAML-" if yaml_available() else "JSON-"
+        parser.add_argument(
+            "--config-file",
+            required=False,
+            type=Path,
+            help=f"Override configuration defaults from a {supported_types}-formatted file.",
```

Comment:
this now results in a double - in the help statement

Reply:
Good one. Fixed